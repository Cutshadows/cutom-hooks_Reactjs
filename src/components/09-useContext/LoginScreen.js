import React,{useContext} from 'react';
import UserContext from './userContext';


const LoginScreen = () => {
    const {user, setUser} = useContext(UserContext);

    return ( 
        <>
            <h1>LoginScreen</h1>
            <hr/>
            <button
            className="btn btn-outline-success"
            onClick={()=>setUser({
                email:'main@email.com',
                name:'Monarca10'
            })}>
                Login
            </button>
        </>
     );
}
 
export default LoginScreen;