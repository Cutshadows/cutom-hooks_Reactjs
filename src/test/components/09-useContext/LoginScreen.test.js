import React from 'react';
import LoginScreen from '../../../components/09-useContext/LoginScreen';
import UserContext from '../../../components/09-useContext/userContext';
import { shallowm, mount } from 'enzyme';



describe('Pruebas sobre <LoginScreen></LoginScreen>', () => {
    
    const setUser=jest.fn();
    const wrapper=mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>    
    )
    test('debe de cargar el componente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de ejecutar el setUser con el argumento esperado  ', () => {
        wrapper.find('button').prop('onClick')();
        expect(setUser).toHaveBeenCalledWith({
            email:'main@email.com',
            name:'Monarca10'
        });
    });



    
})
