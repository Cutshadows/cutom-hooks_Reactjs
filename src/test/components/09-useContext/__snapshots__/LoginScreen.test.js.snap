// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <LoginScreen></LoginScreen> debe de cargar el componente  1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <button
    className="btn btn-outline-success"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
